src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1347,7 +1349,14 @@ private uint getClassAttribsInternal(TypeDesc type)
                     result |= CorInfoFlag.CORINFO_FLG_CONTAINS_GC_PTR;
 
                 if (metadataType.IsBeforeFieldInit)
-                    result |= CorInfoFlag.CORINFO_FLG_BEFOREFIELDINIT;
+                {
+#if READYTORUN
+                    if (_compilation.CompilationModuleGroup.VersionsWithType(metadataType))
```

Comment:
It may be better to clear the flag `CORINFO_FLG_BEFOREFIELDINIT` at the end instead. One less `#if READYTORUN` ifdef and one less call to `VersionsWithType`.

Reply:
Nice, thanks for the suggestion, fixed in 3rd commit.